src/pe/resource.rs

```diff
@@ -0,0 +1,1433 @@
+use crate::error;
+use alloc::string::String;
+use alloc::vec::Vec;
+use core::fmt;
+use core::ops::Not;
+use log::debug;
+use scroll::{Pread, Pwrite, SizeWith};
+
+use crate::pe::data_directories;
+use crate::pe::options;
+use crate::pe::section_table;
+use crate::pe::utils;
+
+/// Size of `wchar_t` in C (aka [`u16`] in Rust)
+pub(super) const SIZE_OF_WCHAR: usize = core::mem::size_of::<u16>();
+/// Converts [`u8`] slice into a vector of [`u16`] and then utf-16 [`String`].
+///
+/// This function assumes that input bytes are multiple of `2`.
+pub(super) fn to_utf16_string(bytes: &[u8]) -> String {
+    let u16_slice = bytes
+        .chunks(2)
+        .map(|chunk| u16::from_le_bytes([chunk[0], chunk[1]]))
+        .take_while(|&wchar| wchar != 0)
+        .collect::<Vec<_>>();
+    String::from_utf16_lossy(&u16_slice)
+}
```

Comment:
It's a lttle big confusing part, but by design (ref `ResourceString::parse`) it is guaranteed to be 2-bytes aligned within usage of this `resource` module. I believe theres better way to do this, or should I remove `alloc::string::String`s and let consumers impl higher implementations (thus keep our impl's lower impls)?